src/librustc/traits/specialize/mod.rs

```diff
@@ -103,6 +105,38 @@ pub fn translate_substs<'a, 'gcx, 'tcx>(infcx: &InferCtxt<'a, 'gcx, 'tcx>,
     source_substs.rebase_onto(infcx.tcx, source_impl, target_substs)
 }
 
+/// Locates the applicable definition of a method, given its name.
```

Comment:
I just moved it from trans, but I suppose I can document it. The `impl_` arguments in particular from `VtableImpl`, I might be able to make it take that type directly.